src/verilog/verilog_typecheck_expr.cpp

```diff
@@ -1545,13 +1545,68 @@ exprt verilog_typecheck_exprt::elaborate_constant_expression(exprt expr)
     if(!operands_are_constant)
       return expr; // give up
 
+    auto make_all_ones = [](const typet &type) -> exprt
+    {
+      if(type.id() == ID_unsignedbv)
+      {
+        return from_integer(
+          power(2, to_unsignedbv_type(type).get_width()) - 1, type);
+      }
+      else if(type.id() == ID_signedbv)
+      {
+        return from_integer(-1, type);
+      }
+      else if(type.id() == ID_bool)
+        return true_exprt{};
+      else
+        PRECONDITION(false);
+    };
```

Comment:
Wouldn't `bv_typet::all_ones_expr` do the trick?

Reply:
Not sure if the simplifier can cast that? Will check.